### PR TITLE
Testing/7 player expansion

### PR DESCRIPTION
**Pull request by ODE-BOY**
# Description
This PR fixes #[284]

Added Domain and Integration tests for 7 player lobbys.
Added 7 tests. 1 Domain, 6 Integration.

# Demo
image

# Type of change:
* New feature (Non-breaking change that adds functionality)

### CODE REVIEW

File: backend/test/integration/GameStartEvents.integration.test.js [hunk 1]
```diff
@@ -34,7 +39,17 @@ describe('GameStartEvents integration tests', () => {
             await connectAndJoin(clientSockets, i, port, lobbyCode);
         }
         /* eslint-enable no-await-in-loop */
-        await startGame(clientSockets);
+        await startGameSixPlayer(clientSockets);
```
Comment (RawiriHohepa) [inline backend/test/integration/GameStartEvents.integration.test.js]: Can you combine the two startGame functions and pass through the number as a parameter?

File: backend/test/integration/StateChangeEvents.integration.test.js [hunk 2]
```diff
@@ -132,7 +175,44 @@ describe('DayStateEvents integration tests', () => {
...
             await connectAndJoin(clientSockets, i, port, lobbyCode);
         }
-        await startGame(clientSockets);
+        await startGameSixPlayer(clientSockets);
+        await startDay();
+
+        done();
+    });
+
+    test('integration test start-day 7 players', async (done) => {
```
Comment (RawiriHohepa) [inline backend/test/integration/StateChangeEvents.integration.test.js]: Is there a way to reduce the duplicate code here and somehow pass in the number of players?

**Review by kyzh0 — DISMISSED** [review]
All tests pass, LGTM!

**Review by RawiriHohepa — CHANGES_REQUESTED** [review]
LGTM, just wondering if there's a way to reduce the amount of duplication, especially if we plan to make more tests with different numbers of players

File: backend/test/integration/LobbyEvents.integration.test.js [hunk 3]
```diff
@@ -25,17 +25,25 @@ describe('lobby events tests', () => {
         SocketIOServer.server.close();
     });
     /**
-     * This is a integration test for all lobby events.
-     * It encompases creating a game, Max players joining and reseting the game
+     * These are a integration tests for all lobby events.
```
Comment (ysway) [inline backend/test/integration/LobbyEvents.integration.test.js]: These are ~~a~~ integration tests for all lobby events.

**Review by ysway — COMMENTED** [review]
grammar

File: backend/test/integration/GameStartEvents.integration.test.js (hunk 1 above)
Comment (ODE-BOY) [reply backend/test/integration/GameStartEvents.integration.test.js]: Good Idea. Ill make those changes.

File: backend/test/integration/LobbyEvents.integration.test.js (hunk 3 above)
Comment (ODE-BOY) [reply backend/test/integration/LobbyEvents.integration.test.js]: Good catch ill fit it up

File: backend/test/integration/StateChangeEvents.integration.test.js (hunk 2 above)
Comment (ODE-BOY) [reply backend/test/integration/StateChangeEvents.integration.test.js]: Done and Done.

File: backend/test/integration/GameStartEvents.integration.test.js [hunk 4]
```diff
@@ -28,13 +28,16 @@ describe('GameStartEvents integration tests', () => {
...
-        /* eslint-disable no-await-in-loop */
-        for (let i = 1; i < 6; i += 1) {
-            await connectAndJoin(clientSockets, i, port, lobbyCode);
-        }
-        /* eslint-enable no-await-in-loop */
-        await startGame(clientSockets);
-        done();
-    });
+    // Run integration tests for all possible lobby sizes.
+    for (let playerCount = 6; playerCount <= 7; playerCount += 1) {
```
Comment (RawiriHohepa) [inline backend/test/integration/GameStartEvents.integration.test.js:32]: This is great, especially making a separate test() for each playerCount. We just need to remember to update these values when the maximum player count changes

**Review by RawiriHohepa — APPROVED** [review]
Looks great, much better without the code duplication

**Review by kyzh0 — APPROVED** [review]
LGTM!

**Pull request merged**